File: AOJ/0289/0289.py
def solve(src, dst):
    if src == dst: return 0
    answer = 0
    if src % 2 != 0:
        answer += 1
        src += 1
    if dst % 2 != 0:
        answer += 1
        dst -= 1
    answer += solve(src // 2, dst // 2)
    return answer

N = int(input())
for i in range(N):
    s,d = map(int, input().split())
    print(solve(s,d))

File: AOJ/0289/0289.cpp
#include<iostream>
using namespace std;

// src から dst までの最短距離を求める
int dist(int src, int dst) {
  if (src == dst) return 0;
  int answer = 0;
  if (src % 2 != 0) { answer++; src++; }
  if (dst % 2 != 0) { answer++; dst--; }
  answer += dist(src / 2, dst / 2);
  return answer;
}

int main(){
  int N, s, d;
  cin >> N;
  for ( int i = 0; i < N; i++ ){
    cin >> s >> d;
    cout << dist(s, d) << endl;
  }
}
